skbase/lookup/_lookup.py

```diff
@@ -233,7 +250,7 @@ def _walk(root, exclude=None, prefix=""):
 
     Parameters
     ----------
-    root : Path
+    root : Path-like
```

Comment:
should we say "i.e., `Path` or `Path` coercible"?

Reply:
I don't feel strongly about this. I like trying to keep type description of param short if possible. But Path or path coercive works.

Reply:
well, either is a bit vague - maybe something more specific, or examples?

Reply:
I went with: str path or path-like. Then in the description portion said it can be a string path, pathlib.Path or other path-like object.